backend/molgenis-emx2-tasks/src/main/java/org/molgenis/emx2/tasks/TaskServiceInMemory.java

```diff
@@ -29,8 +25,18 @@ public List<ScriptTask> getScripts() {
 
   @Override
   public String submit(Task task) {
-    tasks.put(task.getId(), task);
-    executorService.submit(task);
+    if (task.getParentTask() != null && tasks.containsKey(task.getParentTask().getId())) {
+      Task parentTask = tasks.get(task.getParentTask().getId());
+      parentTask.addSubTask(task);
+      if (parentTask.getStatus() == RUNNING) {
+        task.run();
+      } else {
+        executorService.submit(task);
```

Comment:
should not happen? If we have strong subjobbing then the parent job should be running otherwise something is wrong.